Data/Binary/Serialise/CBOR/Read.hs

```diff
@@ -1882,7 +1889,19 @@ readBigNInt bs
 nintegerFromBytes :: BS.ByteString -> Integer
 nintegerFromBytes bs = -1 - uintegerFromBytes bs
 
--- TODO FIXME: optimised implementation for GMP
+#if defined(OPTIMIZE_GMP)
+uintegerFromBytes :: BS.ByteString -> Integer
+uintegerFromBytes (BS.PS fp (I# off#) (I# len#)) =
+  -- This should be safe since we're simply reading from ByteString (which is
+  -- immutable) and GMP allocates a new memory for the Integer, i.e., there is
+  -- no mutation involved.
+  unsafePerformIO $
```

Comment:
This could almost certainly be a `unsafeDupablePerformIO`. It should be safe to duplicate this (in rare circumstances).